server/src/main/java/com/hedera/block/server/persistence/storage/write/BlockAsDirWriterBuilder.java

```diff
@@ -28,38 +29,37 @@
 import java.nio.file.attribute.FileAttribute;
 import java.nio.file.attribute.PosixFilePermission;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
 /**
  * Use builder methods to create a {@link BlockWriter} to write blocks to storage.
  *
  * <p>When a block writer is created, it will provide access to write blocks to storage.
  */
-public class BlockAsDirWriterBuilder {
-
+public final class BlockAsDirWriterBuilder {
     private final BlockNodeContext blockNodeContext;
-    private FileAttribute<Set<PosixFilePermission>> filePerms = FileUtils.defaultPerms;
+    private FileAttribute<Set<PosixFilePermission>> filePerms = DEFAULT_DIR_PERMISSIONS;
```

Comment:
As noted elsewhere, it's never correct to copy a constant like this into a class attribute. Either the caller supplies the value, or it's null.

Reply:
Please see my response for this in the above comments, for the `BlockAsDirWriterBuilder` and `BlockAsDirWriterBuilder`, it applies here as well.